### PR TITLE
Bug fix: Queue mode without TLS cache

### PR DESCRIPTION
**Pull request by jheiskan81**
Fix data message loose when message is triggered from idle state.
Handle properly Server rejected Registration update by 4.3 Forbidden.

### CODE REVIEW

**Comment by jheiskan81** [comment]
@rlubos @SeppoTakalo @VeijoPesonen @simensrostad Please review this bug fix.

File: subsys/net/lib/lwm2m/lwm2m_message_handling.c [hunk 1]
```diff
@@ -204,12 +204,9 @@ void lwm2m_engine_context_close(struct lwm2m_ctx *client_ctx)
 
 void lwm2m_engine_context_init(struct lwm2m_ctx *client_ctx)
 {
-	sys_slist_init(&client_ctx->pending_sends);
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_message_handling.c:207]: `lwm2m_pull_context.c` also creates a `lwm2m_ctx` and makes use of `pending_sends`, IMO those initializers belong here. Perhaps `lwm2m_engine_context_init()` should not be called if the client does not start fresh but wakes up?

**Review by rlubos — COMMENTED** [review]
There are build errors in the CI, please have a look.

File: subsys/net/lib/lwm2m/lwm2m_message_handling.c (hunk 1 above)
Comment (jheiskan81) [reply subsys/net/lib/lwm2m/lwm2m_message_handling.c:207]: OK I will check that also. context init was main key src for loose a buffered data. I will check lwm2m_pull_context.

**Comment by jheiskan81** [comment]
@rlubos Please review. I update PR. I fix also duplicate socket allocation.

File: subsys/net/lib/lwm2m/lwm2m_rd_client.c [hunk 2]
```diff
@@ -1186,6 +1198,10 @@ static void lwm2m_rd_client_service(struct k_work *work)
 			sm_do_registration();
 			break;
 
+		case ENGINE_SEND_REGISTRATION:
+			sm_send_registration_msg();
+		break;
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_rd_client.c:1203]:
```suggestion
			break;
```

**Comment by jheiskan81** [comment]
I have fixed comment's. @rlubos.

**Review by rlubos — APPROVED** [review]

**Review by jarlamsa — APPROVED** [review]

**Review by VeijoPesonen — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Pull request merged**